src/EditorFeatures/Core/GoToDefinition/GoToDefinitionCommandHandler.cs

```diff
@@ -161,32 +163,30 @@ private static void ReportFailure(Document document)
         {
             bool succeeded;
 
-            var indicatorFactory = document.Project.Solution.Services.GetRequiredService<IBackgroundWorkIndicatorFactory>();
-
             // TODO: prior logic was to get a tracking span of length 1 here.  Preserving that, though it's unclear if
             // that is necessary for the BWI to work properly.
             Contract.ThrowIfTrue(position.Snapshot.Length == 0);
             var applicableToSpan = position < position.Snapshot.Length
                 ? new SnapshotSpan(position, position + 1)
                 : new SnapshotSpan(position - 1, position);
 
-            using (var backgroundIndicator = indicatorFactory.Create(
+            using (var backgroundIndicator = _backgroundWorkIndicatorService.Create(
                 args.TextView, applicableToSpan,
                 EditorFeaturesResources.Navigating_to_definition))
             {
-                var cancellationToken = backgroundIndicator.UserCancellationToken;
+                var cancellationToken = backgroundIndicator.CancellationToken;
 
                 // determine the location first.
                 var (location, _) = await service.FindDefinitionLocationAsync(
                     document, position, includeType: true, cancellationToken).ConfigureAwait(false);
 
                 // make sure that if our background indicator got canceled, that we do not still perform the navigation.
-                if (backgroundIndicator.UserCancellationToken.IsCancellationRequested)
+                if (backgroundIndicator.CancellationToken.IsCancellationRequested)
```

Comment:
Can this be simplified?

```suggestion
                if (cancellationToken)
```